athletics.lic

```diff
@@ -0,0 +1,223 @@
+custom_require.call(%w(common common-travel))
+
+class Athletics
+  
+  include DRC
+  include DRCT
+  
+  def initialize
+    @did_play = false
+    @no_instrument = false
+    @settings = get_settings()
+    @athletics_options = get_data('athletics').athletics_options
+    @song_list = get_data('perform').perform_options
+ 
+    Flags.add('ct-song', 'you finish playing')
+    play_song?
+    
+    @start_skill=DRSkill.getxp('Athletics')
+    @end_skill=@start_skill+20
+    if(@end_skill>30)
+      @end_skill=30
+    end
+
+    if @settings.hometown == 'Crossing' || @settings.hometown == 'Riverhaven'
+      crossing_athletics
+    elsif @settings.hometown == 'Shard'
+      if DRSkill.getrank('Athletics') <= 250
+        shard_athletics
+      else
+        if DRSkill.getrank('Stealth') >= 250
+          climb_practice(19460,"wall",true)
+        else
+          hib_athletics
+        end
+      end
+    elsif @settings.hometown == 'Hibarnhvidar'
+      hib_athletics
+    elsif @settings.hometown == 'Ratha'
+      ratha_athletics
+    end
+  end
+  
+  def crossing_athletics
+    if DRSkill.getrank('Athletics') >= 400
+      return if @researching
+      climb_branch
+    else
+      while DRSkill.getxp('Athletics') < @end_skill
+        @athletics_options
+          .reject { |data| @settings.avoid_athletics_in_justice && data['justice'] }
+          .each do |data|
+            break unless climb?(data['room'], data['targets'])
+        end
+      end
+    end
+  end
+  
+  def shard_athletics
+    while DRSkill.getxp('Athletics') <= @end_skill
+      walk_to(11481)
+      walk_to(11482)
+      walk_to(11483)
+    end    
+    
+    walk_to(@settings.safe_room)
```

Comment:
What motivated this walk back? If we allow this script more run time flexibility this seems like a problem and would be better managed by another before/after with a go2 or as part of crossing-training.